File: src/main/java/com/jakewharton/sdksearch/Item.java
package com.jakewharton.sdksearch;

final class Item {
  final long id;
  final String label;
  final String link;
  final String type;

  Item(long id, String label, String link, String type) {
    this.id = id;
    this.label = label;
    this.link = link;
    this.type = type;
  }

  @Override public String toString() {
    return "Item{"
        + "id="
        + id
        + ", label='"
        + label
        + '\''
        + ", link='"
        + link
        + '\''
        + ", type='"
        + type
        + '\''
        + '}';
  }
}

File: api/dac/model/src/main/kotlin/com/jakewharton/sdksearch/api/dac/model/Item.kt
package com.jakewharton.sdksearch.api.dac.model

import kotlinx.serialization.KInput
import kotlinx.serialization.KOutput
import kotlinx.serialization.KSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.Serializer

@Serializable
data class Item(
  val id: Int,
  val label: String,
  val link: String,
  val type: String,
  @Serializable(with = BooleanStringSerializer::class) val deprecated: Boolean
)

@Serializer(forClass = Boolean::class)
object BooleanStringSerializer : KSerializer<Boolean> {
  override fun save(output: KOutput, obj: Boolean) = output.writeStringValue(obj.toString())
  // TODO https://youtrack.jetbrains.com/issue/KT-16348
  override fun load(input: KInput) = input.readStringValue().toLowerCase() == "true"
}
